src/llm/language_model/legacy/servable.cpp

```diff
@@ -0,0 +1,152 @@
+        return status;
+    }
+
+    if (legacyExecutionContext->apiHandler->isStream()) {
+        legacyExecutionContext->lastStreamerCallbackOutput = "";  // initialize with empty string
+        auto callback = [& mutex = legacyExecutionContext->mutex, &lastStreamerCallbackOutput = legacyExecutionContext->lastStreamerCallbackOutput](std::string text) {
+            SPDLOG_LOGGER_TRACE(llm_calculator_logger, "Streamer callback executed with text: [{}]", text);
+            std::lock_guard<std::mutex> lock(mutex);
+            lastStreamerCallbackOutput += text;
+            return ov::genai::StreamingStatus::RUNNING;
+        };
+        legacyExecutionContext->textStreamer = std::make_shared<ov::genai::TextStreamer>(getProperties()->tokenizer, callback);
+    }
+    return absl::OkStatus();
+}
+
+absl::Status LegacyServable::scheduleExecution(std::shared_ptr<GenAiServableExecutionContext>& executionContext) {
+    auto legacyExecutionContext = std::static_pointer_cast<LegacyServableExecutionContext>(executionContext);
+    if (legacyExecutionContext->payload.client->isDisconnected()) {
+        return absl::CancelledError();
+    }
+    properties->legacyExecutor->addRequest(legacyExecutionContext);
+    return absl::OkStatus();
+}
+
+absl::Status LegacyServable::readCompleteExecutionResults(std::shared_ptr<GenAiServableExecutionContext>& executionContext) {
+    auto legacyExecutionContext = std::static_pointer_cast<LegacyServableExecutionContext>(executionContext);
+    if (legacyExecutionContext->payload.client->isDisconnected()) {
+        return absl::CancelledError();
+    }
+    legacyExecutionContext->finished.wait();
+    return absl::OkStatus();
+}
+
+absl::Status LegacyServable::prepareCompleteResponse(std::shared_ptr<GenAiServableExecutionContext>& executionContext) {
+    auto legacyExecutionContext = std::static_pointer_cast<LegacyServableExecutionContext>(executionContext);
+    if (legacyExecutionContext->payload.client->isDisconnected()) {
+        return absl::CancelledError();
+    }
+    executionContext->response = executionContext->apiHandler->serializeUnaryResponse(legacyExecutionContext->results);
+    SPDLOG_LOGGER_DEBUG(llm_calculator_logger, "Complete unary response: {}", executionContext->response);
+    return absl::OkStatus();
+}
+
+absl::Status LegacyServable::readPartialExecutionResults(std::shared_ptr<GenAiServableExecutionContext>& executionContext) {
+    return absl::OkStatus();
+}
+
+absl::Status LegacyServable::preparePartialResponse(std::shared_ptr<GenAiServableExecutionContext>& executionContext) {
+    auto legacyExecutionContext = std::static_pointer_cast<LegacyServableExecutionContext>(executionContext);
+    if (legacyExecutionContext->payload.client->isDisconnected()) {
+        return absl::CancelledError();
+    }
+    std::stringstream ss;
+    ss << executionContext->lastStreamerCallbackOutput;
+    std::string lastTextChunk = ss.str();
+    legacyExecutionContext->mutex.lock();
```

Comment:
I think it would be safer to move this lock above 
`ss << executionContext->lastStreamerCallbackOutput;`
also, what do you think about closing this logic in a scope and using scoped_lock instead of lock() and unlock() calls?